neo/rawio/openephysbinaryrawio.py

```diff
@@ -349,72 +355,155 @@ def explore_folder(dirname):
                 node_name = ''
 
-            block_index = int(root.parents[0].stem.lower().replace('experiment', '')) - 1
-            if block_index not in all_streams:
-                all_streams[block_index] = {}
-                nb_block += 1
-                nb_segment_per_block[block_index] = 0
-
-            seg_index = int(root.stem.replace('recording', '')) - 1
-            if seg_index not in all_streams[block_index]:
-                all_streams[block_index][seg_index] = {
-                    'continuous': {},
-                    'events': {},
-                    'spikes': {},
-                }
-                if seg_index >= nb_segment_per_block[block_index]:
-                    nb_segment_per_block[block_index] = seg_index + 1
-
-            # metadata
-            with open(root / 'structure.oebin', encoding='utf8', mode='r') as f:
-                structure = json.load(f)
-
-            if (root / 'continuous').exists() and len(structure['continuous']) > 0:
-                for d in structure['continuous']:
-                    # when multi Record Node the stream name also contains
-                    # the node name to make it unique
-                    stream_name = node_name + '#' + d['folder_name']
-
-                    raw_filename = root / 'continuous' / d['folder_name'] / 'continuous.dat'
-
-                    timestamp_file = root / 'continuous' / d['folder_name'] / 'timestamps.npy'
-                    timestamps = np.load(str(timestamp_file), mmap_mode='r')
-                    timestamp0 = timestamps[0]
-                    t_start = timestamp0 / d['sample_rate']
-
-                    # TODO for later : gap checking
-                    signal_stream = d.copy()
-                    signal_stream['raw_filename'] = str(raw_filename)
-                    signal_stream['dtype'] = 'int16'
-                    signal_stream['timestamp0'] = timestamp0
-                    signal_stream['t_start'] = t_start
-
-                    all_streams[block_index][seg_index]['continuous'][stream_name] = signal_stream
-
-            if (root / 'events').exists() and len(structure['events']) > 0:
-                for d in structure['events']:
-                    stream_name = node_name + '#' + d['folder_name']
-
-                    event_stream = d.copy()
-                    for name in _possible_event_stream_names:
-                        npz_filename = root / 'events' / d['folder_name'] / f'{name}.npy'
-                        if npz_filename.is_file():
-                            event_stream[f'{name}_npy'] = str(npz_filename)
-
-                    all_streams[block_index][seg_index]['events'][stream_name] = event_stream
-
-    # TODO for later: check stream / channel consistency across segment
-
-    return all_streams, nb_block, nb_segment_per_block
+            if node_name not in folder_structure:
+                folder_structure[node_name] = {}
+                folder_structure[node_name]['experiments'] = []
```

Comment:
Do we naturally sort this list after exploring the folder ?

Reply:
mmm maybe we should actually!

Reply:
@samuelgarcia done: both for experiments (block_index) and for recordings (seg_index)